PR: https://github.com/odoo/odoo/pull/191963

From: fc5f31b5882d78d9294e420af5e6a9258d407539
From: Gorash
Date: 2024-12-28 08:04:34

Breaking data model changes scores: del:3 + add:0, change matches:
-    _inherit = ['ir.actions.server']
-    _inherit = ['ir.qweb']
-    _inherit = ['ir.model.fields']

Total Changes: 8

[IMP] orm/all: Make _name or _inherit mandatory

This is a request following the revert of the use of class names
(concerning the python inheritance and typing project)

closes odoo/odoo#191963

X-original-commit: d6bf6df30095a27eac0315ea78bb57b721efdd5e
Related: odoo/enterprise#76269
Related: odoo/design-themes#1040
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/website/models/ir_actions_server.py
+++ b/addons/website/models/ir_actions_server.py
@@ -9,7 +9,7 @@ from odoo.tools.json import scriptsafe as json_scriptsafe
 class IrActionsServer(models.Model):
     """ Add website option in server actions. """
 
-    _inherit = ['ir.actions.server']
+    _inherit = 'ir.actions.server'
 
     xml_id = fields.Char('External ID', compute='_compute_xml_id', help="ID of the action if defined in a XML file")
     website_path = fields.Char('Website Path')

--- a/addons/website/models/ir_qweb.py
+++ b/addons/website/models/ir_qweb.py
@@ -21,7 +21,7 @@ re_background_image = re.compile(r"(background-image\s*:\s*url\(\s*['\"]?\s*)([^
 class IrQweb(models.AbstractModel):
     """ IrQweb object for rendering stuff in the website context """
 
-    _inherit = ['ir.qweb']
+    _inherit = 'ir.qweb'
 
     URL_ATTRS = {
         'form': 'action',

--- a/addons/website/models/mixins.py
+++ b/addons/website/models/mixins.py
@@ -234,6 +234,7 @@ class WebsitePublishedMixin(models.AbstractModel):
 
 
 class WebsitePublishedMultiMixin(WebsitePublishedMixin):
+    _name = 'website.published.multi.mixin'
     _inherit = ['website.published.mixin', 'website.multi.mixin']
     _description = 'Multi Website Published Mixin'
 
@@ -289,6 +290,7 @@ class WebsitePublishedMultiMixin(WebsitePublishedMixin):
 
 class WebsiteSearchableMixin(models.AbstractModel):
     """Mixin to be inherited by all models that need to searchable through website"""
+    _name = 'website.searchable.mixin'
     _description = 'Website Searchable Mixin'
 
     @api.model

--- a/addons/website/models/website_form.py
+++ b/addons/website/models/website_form.py
@@ -131,7 +131,7 @@ class IrModel(models.Model):
 class IrModelFields(models.Model):
     """ fields configuration for form builder """
     _description = 'Fields'
-    _inherit = ['ir.model.fields']
+    _inherit = 'ir.model.fields'
 
     def init(self):
         # set all existing unset website_form_blacklisted fields to ``true``
